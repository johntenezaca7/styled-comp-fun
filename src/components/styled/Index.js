import React from "react";
import styled, { createGlobalStyle  } from "styled-components";

export { Button } from "./Elements/Button";
export { Title } from "./Elements/Title";
export { Section } from "./Elements/Section";

export const Theme = {
  main: "mediumseagreen"
};

export const DivWrapper = styled.div`
  border: #70A288 2px solid;
  padding 2rem;
  margin: 1rem 0;
`;

export const GlobalStyle = createGlobalStyle`
  body {
    @import url('https://fonts.googleapis.com/css?family=Exo&display=swap');
    padding: 0;
    margin: 0;
    font-family: 'Exo', sans-serif;

    #root  {
      margin: 0;
    }
  }
`;