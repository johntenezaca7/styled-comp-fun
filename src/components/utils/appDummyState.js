const state = {
    items: [
        {
          id: 'a',
          details: {
            name: "Poo",
            animal: "dog",
            breed: "poodle"
          }
        },
        {
          id: 'b',
          details: {
            name: "Yeet",
            animal: "Cat",
            breed: "Street"
          }
        },
        {
          id: 'c',
          details: {
            name: "Skoot",
            animal: "dog",
            breed: "poodle"
          }
        },
        {
          id: 'd',
          details: {
            name: "Lip",
            animal: "dog",
            breed: "poodle"
          }
        },
    ]
};

export { state as default };