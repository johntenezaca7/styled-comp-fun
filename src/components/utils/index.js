export { tupleNestedArray } from "./tupleNestedArray";
export { flattenNestObj } from "./FlattenNestObj";