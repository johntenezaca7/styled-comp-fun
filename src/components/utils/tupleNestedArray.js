// Take two arrays and create a touple at each index
// Example:
// var arr1 = ['a', 'b', 'c']
// var arr2 = ['1']

// Use FlattenNestObj before using this

const tupleNestedArray = (obj) => {
  const res = [];
  for ( const key in obj ) {
    res.push([key, obj[key]]);
  }
  return res
};

export { tupleNestedArray };