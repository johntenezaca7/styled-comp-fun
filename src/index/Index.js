import React from "react";
import { render } from "react-dom";
import Init from "./Init";

render( <Init /> , document.getElementById("root"));