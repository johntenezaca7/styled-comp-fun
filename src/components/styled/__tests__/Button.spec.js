import React from "react";
import renderer from "react-test-renderer";
import { tupleNestedArray, flattenNestObj } from "../../utils";
import "jest-styled-components";

import { Button } from "../Index";

const DefaultProps = tupleNestedArray( flattenNestObj(Button.defaultProps) );

test("Button renders correctly.", () => {
  const element = renderer.create(<Button />).toJSON();
  expect(element).toMatchSnapshot();
});

DefaultProps.forEach(prop => {
  test(`Default props: ${prop[0]} should be ${prop[1]}`, () => {
    const element = renderer.create(<Button />).toJSON();
    expect(element).toHaveStyleRule(prop[0], prop[1]);
  });
});