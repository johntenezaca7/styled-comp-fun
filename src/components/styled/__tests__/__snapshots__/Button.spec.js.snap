// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly. 1`] = `
.c0 {
  font-size: 1em;
  border-radius: 3px;
  cursor: pointer;
  color: #70A288;
  border: 2px solid #70A288;
  padding: 0.25rem 1rem;
  margin: 0;
}

<button
  className="c0"
/>
`;
